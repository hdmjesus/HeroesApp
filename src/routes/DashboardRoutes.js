import React from "react";
import { Route, Switch, Redirect } from "react-router-dom";
import { NavBar } from "../Components/ui/NavBar";
import { DcScreen } from "../Screen/DcScreen";
import { HeroesScreen } from "../Screen/HeroesScreen";
import { MarvelScreen } from "../Screen/MarvelScreen";

export const DashboardRoutes = () => {
  return (
    <>
      <NavBar />
      <div>
        <Switch>
          <Route exact path="/marvel" component={MarvelScreen} />
          <Route exact path="/heroe/:heroid" component={HeroesScreen} />
          <Route exact path="/dc" component={DcScreen} />
          <Redirect to="/marvel" />
        </Switch>
      </div>
    </>
  );
};
