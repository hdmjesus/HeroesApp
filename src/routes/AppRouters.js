import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";

import { LoginScreen } from "../Screen/LoginScreen";
import { DashboardRoutes } from "./DashboardRoutes";

const App = () => {
  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/login" component={LoginScreen} />
        <Route path="/" component={DashboardRoutes} />
      </Switch>
    </BrowserRouter>
  );
};

export default App;
