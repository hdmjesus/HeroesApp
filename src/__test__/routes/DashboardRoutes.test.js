import React from "react";
import { DashboardRoutes } from "../../routes/DashboardRoutes";
import { mount, shallow } from "enzyme";
import { MemoryRouter } from "react-router";
import { AppRouters } from "../../routes/AppRouters";
import { AuthContext } from "../../Context/AuthContext";

describe("Pruebas en <DashboardRoutes/>", () => {
  const { Provider } = AuthContext;
  const contextValue = {
    dispatch: jest.fn(),
    user: { logged: true, name: "Jesus Hernandez" || expect.any(String) },
  };
  test("should show good ", () => {
    const wrapper = mount(
      <Provider value={contextValue}>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".text-info").text().trim()).toBe("Jesus Hernandez");
  });
});
