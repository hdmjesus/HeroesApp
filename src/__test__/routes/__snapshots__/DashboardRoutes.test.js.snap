// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRoutes/> should show good  1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark px-5"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      HeroesApp
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Jesus Hernandez
        </span>
        <button
          className="btn text-white "
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <div
      className="container mt-5"
    >
      <h1
        className="m-3"
      >
        Marvel Comics
      </h1>
      <hr />
      <div
        className="flex justify-center flex-wrap animate__animated animate__fadeIn"
      >
        <div
          className="card mt-5 mb-3 m-1  text-xs"
          style={
            Object {
              "maxWidth": 350,
              "minWidth": 150,
              "padding": 0,
            }
          }
        >
          <div
            className="row g-0"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="superhero"
                className="card-img"
                src="./images/marvel-spider.webp"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Spider Man
                </h5>
                <p
                  className="card-text"
                >
                  Peter Parker
                </p>
                <p
                  className="card-text"
                >
                   
                  <small
                    className="text-muted"
                  >
                     
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  className="text-blue-500"
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card mt-5 mb-3 m-1  text-xs"
          style={
            Object {
              "maxWidth": 350,
              "minWidth": 150,
              "padding": 0,
            }
          }
        >
          <div
            className="row g-0"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="superhero"
                className="card-img"
                src="./images/marvel-captain.webp"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Captain America
                </h5>
                <p
                  className="card-text"
                >
                  Steve Rogers
                </p>
                <p
                  className="card-text"
                >
                   
                  <small
                    className="text-muted"
                  >
                     
                    Captain America Comics #1
                  </small>
                </p>
                <a
                  className="text-blue-500"
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card mt-5 mb-3 m-1  text-xs"
          style={
            Object {
              "maxWidth": 350,
              "minWidth": 150,
              "padding": 0,
            }
          }
        >
          <div
            className="row g-0"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="superhero"
                className="card-img"
                src="./images/marvel-iron.webp"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Iron Man
                </h5>
                <p
                  className="card-text"
                >
                  Tony Stark
                </p>
                <p
                  className="card-text"
                >
                   
                  <small
                    className="text-muted"
                  >
                     
                    Tales of Suspense #39
                  </small>
                </p>
                <a
                  className="text-blue-500"
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card mt-5 mb-3 m-1  text-xs"
          style={
            Object {
              "maxWidth": 350,
              "minWidth": 150,
              "padding": 0,
            }
          }
        >
          <div
            className="row g-0"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="superhero"
                className="card-img"
                src="./images/marvel-thor.webp"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Thor
                </h5>
                <p
                  className="card-text"
                >
                  Thor Odinson
                </p>
                <p
                  className="card-text"
                >
                   
                  <small
                    className="text-muted"
                  >
                     
                    Journey into Myster #83
                  </small>
                </p>
                <a
                  className="text-blue-500"
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card mt-5 mb-3 m-1  text-xs"
          style={
            Object {
              "maxWidth": 350,
              "minWidth": 150,
              "padding": 0,
            }
          }
        >
          <div
            className="row g-0"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="superhero"
                className="card-img"
                src="./images/marvel-hulk.webp"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hulk
                </h5>
                <p
                  className="card-text"
                >
                  Bruce Banner
                </p>
                <p
                  className="card-text"
                >
                   
                  <small
                    className="text-muted"
                  >
                     
                    The Incredible Hulk #1
                  </small>
                </p>
                <a
                  className="text-blue-500"
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card mt-5 mb-3 m-1  text-xs"
          style={
            Object {
              "maxWidth": 350,
              "minWidth": 150,
              "padding": 0,
            }
          }
        >
          <div
            className="row g-0"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="superhero"
                className="card-img"
                src="./images/marvel-wolverine.webp"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Wolverine
                </h5>
                <p
                  className="card-text"
                >
                  James Howlett
                </p>
                <p
                  className="card-text"
                >
                   
                  <small
                    className="text-muted"
                  >
                     
                    The Incredible Hulk #180
                  </small>
                </p>
                <a
                  className="text-blue-500"
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card mt-5 mb-3 m-1  text-xs"
          style={
            Object {
              "maxWidth": 350,
              "minWidth": 150,
              "padding": 0,
            }
          }
        >
          <div
            className="row g-0"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="superhero"
                className="card-img"
                src="./images/marvel-daredevil.webp"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Daredevil
                </h5>
                <p
                  className="card-text"
                >
                  Matthew Michael Murdock
                </p>
                <p
                  className="card-text"
                >
                   
                  <small
                    className="text-muted"
                  >
                     
                    Daredevil #1
                  </small>
                </p>
                <a
                  className="text-blue-500"
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card mt-5 mb-3 m-1  text-xs"
          style={
            Object {
              "maxWidth": 350,
              "minWidth": 150,
              "padding": 0,
            }
          }
        >
          <div
            className="row g-0"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="superhero"
                className="card-img"
                src="./images/marvel-hawkeye.webp"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hawkeye
                </h5>
                <p
                  className="card-text"
                >
                  Clinton Francis Barton
                </p>
                <p
                  className="card-text"
                >
                   
                  <small
                    className="text-muted"
                  >
                     
                    Tales of Suspense #57
                  </small>
                </p>
                <a
                  className="text-blue-500"
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card mt-5 mb-3 m-1  text-xs"
          style={
            Object {
              "maxWidth": 350,
              "minWidth": 150,
              "padding": 0,
            }
          }
        >
          <div
            className="row g-0"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="superhero"
                className="card-img"
                src="./images/marvel-cyclops.webp"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Cyclops
                </h5>
                <p
                  className="card-text"
                >
                  Scott Summers
                </p>
                <p
                  className="card-text"
                >
                   
                  <small
                    className="text-muted"
                  >
                     
                    X-Men #1
                  </small>
                </p>
                <a
                  className="text-blue-500"
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card mt-5 mb-3 m-1  text-xs"
          style={
            Object {
              "maxWidth": 350,
              "minWidth": 150,
              "padding": 0,
            }
          }
        >
          <div
            className="row g-0"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="superhero"
                className="card-img"
                src="./images/marvel-silver.webp"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Silver Surfer
                </h5>
                <p
                  className="card-text"
                >
                  Norrin Radd
                </p>
                <p
                  className="card-text"
                >
                   
                  <small
                    className="text-muted"
                  >
                     
                    The Fantastic Four #48
                  </small>
                </p>
                <a
                  className="text-blue-500"
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
